code/datums/diseases/advance/symptoms/heal.dm

```diff
@@ -618,3 +618,412 @@ im not even gonna bother with these for the following symptoms. typed em out, co
+				M.blood_volume = (M.blood_volume - 4)
+				bloodpoints += 1
+			else
+				bloodpoints += max(0, grabbedblood)
+			for(var/I in 1 to power)//power doesnt increase efficiency, just usage. 
+				if(bloodpoints)
+					if(H.bleed_rate >= 2 && bruteheal && bloodpoints)
+						bloodpoints -= 1
+						H.bleed_rate = max(0, (H.bleed_rate - 2))
+					if(M.blood_volume < BLOOD_VOLUME_NORMAL && M.get_blood_id() == /datum/reagent/blood) //bloodloss is prioritized over healing brute
+						bloodpoints -= 1
+						M.blood_volume = max((M.blood_volume + 3 * power), BLOOD_VOLUME_NORMAL) //bloodpoints are valued at 4 units of blood volume per point, so this is diminished
+					else if(bruteheal && M.getBruteLoss())
+						bloodpoints -= 1
+						M.heal_overall_damage(2, required_status = BODYPART_ORGANIC)
+					if(prob(20) && !M.stat) 
+						bloodpoints -- //you cant just accumulate blood and keep it as a battery of healing. the quicker the symptom is, the faster your bloodpoints decay
+				else if(prob(60) && M.blood_volume >= BLOOD_VOLUME_BAD)//the virus continues to extract blood if you dont have any stored up. higher probability due to BP value
+					M.blood_volume = (M.blood_volume -= 1)
+
+			if(!bloodpoints && prob(3))
+				to_chat(M, "<span class='warning'>[pick("You feel a pang of thirst.", "No food can sate your hunger", "Blood...")]</span>")
+
+/datum/symptom/vampirism/End(datum/disease/advance/A)
+	. = ..()
+	REMOVE_TRAIT(A.affected_mob, TRAIT_DRINKSBLOOD, DISEASE_TRAIT)
+	if(bloodtypearchive && ishuman(A.affected_mob))
+		var/mob/living/carbon/human/H = A.affected_mob
+		H.dna.blood_type = bloodtypearchive 
+
+/datum/symptom/vampirism/proc/succ(mob/living/carbon/M) //you dont need the blood reagent to suck blood. however, you need to have blood, or at least a shared blood reagent, for most of the other uses
+	var/gainedpoints = 0
+	if(bloodbag && !bloodbag.blood_volume) //we've exsanguinated them!
+		bloodbag = null
+	if(ishuman(M) && M.stat == DEAD && vampire)
+		var/mob/living/carbon/human/H = M
+		var/possibledist = power + 1
+		if(M.get_blood_id() != /datum/reagent/blood)
+			possibledist = 1
+		if(!(NOBLOOD in H.dna.species.species_traits)) //if you dont have blood, well... sucks to be you
+			if(bloodpoints >= 200 && H.health > 0 && H.blood_volume >= BLOOD_VOLUME_NORMAL) //note that you need to actually need to heal, so a maxed out virus won't be bringing you back instantly in most cases. *even so*, if this needs to be nerfed ill do it in a heartbeat
+				H.revive(0)
+				H.visible_message("<span class='warning'>[H.name]'s skin takes on a rosy hue as they begin moving. They live again!</span>", "<span class='userdanger'>As your body fills with fresh blood, you feel your limbs once more, accompanied by an insatiable thirst for blood.</span>")
+				bloodpoints -= 200
+				return 0
+			else if(bloodbag && bloodbag.blood_volume && (bloodbag.stat || bloodbag.bleed_rate))
+				if(get_dist(bloodbag, H) <= 1)
```

Comment:
This seems *incredibly* unlikely, but could you perhaps somehow be sucking somebody standing on another Z level at the same coordinates? `get_dist` considers Z levels as having a distance of 1, after all.

Reply:
I'm not sure if they could be added as a candidate in the first place. Maybe? it seems a bit niche to be something that needs to be worried about much. i cant forsee them being sucked through the floor...

Reply:
is this why you've been waiting so long? didnt even see this under the other review comments, fixed